src/fast_check/transform.rs

```diff
@@ -1463,27 +1668,260 @@ impl<'a> FastCheckTransformer<'a> {
     Ok(is_leavable)
   }
 
-  fn maybe_infer_type_from_expr(&self, expr: &Expr) -> Option<TsType> {
+  fn maybe_infer_type_from_expr(
+    &mut self,
+    expr: &Expr,
+    report_diagnostics: bool,
+  ) -> Result<Option<TsType>, Vec<FastCheckDiagnostic>> {
```

Comment:
Can we separate out the inference into a separate PR (or it looks like multiple separate PRs)? Then we can really make sure we get a good end result for each of them. Doing inference is kind of risky with fast check.